dex/market.go

```diff
@@ -5,17 +5,19 @@ package dex
 
 import (
 	"fmt"
+	"math"
 	"strings"
 )
 
 // MarketInfo specifies a market that the Archiver must support.
 type MarketInfo struct {
-	Name            string
-	Base            uint32
-	Quote           uint32
-	LotSize         uint64
-	EpochDuration   uint64 // msec
-	MarketBuyBuffer float64
+	Name                   string
+	Base                   uint32
+	Quote                  uint32
+	LotSize                uint64
+	EpochDuration          uint64 // msec
+	MarketBuyBuffer        float64
+	MaxUserCancelsPerEpoch uint32
```

Comment:
This is in `MarketInfo` only because it's a convenient way to pass this config to `NewMarket`.  The `MarketInfo` constructors do not take the setting as an input.  We can rethink how we handle this setting if it does prove useful (@behindtext mentioned a limit on max cancels per epoch per user is desirable, and I tend to agree).